File: demo/demo/ViewController.m
//
//  ViewController.m
//  demo
//
//  Created by 跑酷 on 15/8/11.
//  Copyright (c) 2015年 nonato. All rights reserved.
//
#import "SystyleviewController.h"
#import "ViewController.h"
#import "HBTable.h"
#import "TESTXIBViewController.h"

@interface ViewController ()
//step 1
AS_CELL_STRUCT_COMMON(sys)
AS_CELL_STRUCT_COMMON(xib)
AS_CELL_STRUCT_COMMON(autoheight)
AS_CELL_STRUCT_COMMON(refresh)

@end

@implementation ViewController
//step 2
GET_CELL_STRUCT_WITH(sys, HBTABLE-系统控件)
GET_CELL_STRUCT_WITH(xib, HBTABLE-加载XIB)
GET_CELL_STRUCT_WITH(autoheight, HBTABLE-自动高度)
GET_CELL_STRUCT_WITH(refresh, HBTABLE-上下拉)

/**
 *  响应的CELselect的方法
 */
//step 3
GET_CELL_SELECT_ACTION(cellstruct)
{
    if(cellstruct == self.cell_struct_sys)
    {
        SystyleviewController * ctr = [[SystyleviewController alloc] init];
        [self.navigationController pushViewController:ctr animated:YES];
    }
    else if(cellstruct == self.cell_struct_xib)
    {
        TESTXIBViewController * ctr = [[TESTXIBViewController alloc] init];
        [self.navigationController pushViewController:ctr animated:YES];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"HBTABLE";
    NSInteger rowIndex = 0;
    [self.dataDictionary setObject:self.cell_struct_sys forKey:KEY_INDEXPATH(0, rowIndex++)];
    [self.dataDictionary setObject:self.cell_struct_xib forKey:KEY_INDEXPATH(0, rowIndex++)];
    [self.dataDictionary setObject:self.cell_struct_autoheight forKey:KEY_INDEXPATH(0, rowIndex++)];
  
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: FastTableView_Swift/FastTableView_Swift/ViewController.swift
//
//  ViewController.swift
//  FastTableView_Swift
//
//  Created by zeno on 15/10/9.
//  Copyright © 2015年 nonato. All rights reserved.
//

import UIKit
import Alamofire

class ViewController: HB_BaseTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
         
        print("1")
        dispatch_async(dispatch_get_main_queue()) { () -> Void in
            print("2")
        }
        print("3")
        
       self.TABLEVIEW_REGISTERXIBCELL_CLASS("TestTableViewCell")
        
        for  index in 0...10
        {
            let title = "title" + String(index)
            let cellstruct6:CELL_STRUCT! = CELL_STRUCT.init(title: title)
            cellstruct6.cellclass = "TestTableViewCell"
            cellstruct6.xibvalue = "xib"
            self.dataDictionary[self.KEY_INDEXPATH(0, ROW: index)] = cellstruct6
        }
//        [super viewDidLoad];
//        NSLog(@"1");
//        dispatch_sync(dispatch_get_main_queue(), ^{
//        NSLog(@"2");
//        });
//        NSLog(@"3"); 
        
        
        //Alamofire.request(.GET, "http://httpbin.org/get")
        
//        Alamofire.request(.GET, "http://httpbin.org/get", parameters: ["foo": "bar"])
//            .response { request, response, data, error in
//                print("request = \(request)")
//                print("response =\(response)")
//                print("error = \(error)")
//        }
        
       Alamofire.request(.GET, "http://httpbin.org/get").responseString(encoding: NSUTF8StringEncoding) { (request, response, result) -> Void in
            print("request = \(request)")
            print("response =\(response)")
            print("result = \(result)")
        }
//            .responseString { _, _, string, _ in
//                print(string)
//            }
//            .responseJSON { _, _, JSON, _ in
//                print(JSON)
//        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

